Riot/Modules/Room/Views/URLPreviews/URLPreviewView.swift

```diff
@@ -0,0 +1,191 @@
+// 
+// Copyright 2021 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import UIKit
+import Reusable
+
+@objc
+protocol URLPreviewViewDelegate: AnyObject {
+    func didOpenURLFromPreviewView(_ previewView: URLPreviewView, for eventID: String, in roomID: String)
+    func didCloseURLPreviewView(_ previewView: URLPreviewView, for eventID: String, in roomID: String)
+}
+
+@objcMembers
+/// A view to display `URLPreviewData` generated by the `URLPreviewManager`.
+class URLPreviewView: UIView, NibLoadable, Themable {
+    // MARK: - Constants
+    
+    private static let sizingView = URLPreviewView.instantiate()
+    
+    private enum Constants {
+        /// The fixed width of the preview view.
+        static let width: CGFloat = 267.0
```

Comment:
The magic number for URL Previews 😆

Reply:
I've been seen this number in my sleep 😅